cfme/tests/automate/test_domain.py

```diff
@@ -514,3 +516,57 @@ def test_automate_restrict_domain_crud(user, custom_instance):
         assert not view.configuration.is_displayed
         view = navigate_to(instance.klass.namespace.domain, "Details")
         assert not view.configuration.is_displayed
+
+
+@pytest.mark.tier(2)
+@pytest.mark.long_running
+@pytest.mark.meta(automates=[1693362])
+@pytest.mark.parametrize("file_name", ["test_migration_db_510.backup"])
+def test_redhat_domain_sync_after_upgrade(temp_appliance_preconfig, file_name):
+    """
+    Bugzilla:
+        1693362
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseposneg: positive
+        casecomponent: Automate
+        testSteps:
+            1. Either dump database of appliance with version X to appliance with version Y
+               or upgrade the appliance
+            2. grep 'domain version on disk differs from db version' /var/www/miq/vmdb/log/evm.log
+            3. Check last_startup.txt file
+        expectedResults:
+            1.
+            2. You should find this string in logs: RedHat domain version on disk differs from db
+               version
+            3. You should find this string in file: RedHat domain version on disk differs from db
+               version
+    """
+    db_file = FTPClientWrapper(cfme_data.ftpserver.entities.databases).get_file(file_name)
+    db_path = f"/tmp/{db_file.name}"
+
+    # Download the customer db on appliance
+    result = temp_appliance_preconfig.ssh_client.run_command(
+        f"curl -o {db_path} ftp://{db_file.link}"
+    )
+    assert result.success
+
+    def _check_file_size(file_path, expected_size):
+        return temp_appliance_preconfig.ssh_client.run_command(
+            f"stat {file_path} | grep {expected_size}"
+        ).success
+
+    # assert the whole database was downloaded to make sure we do not test with broken database
+    assert _check_file_size(db_path, db_file.filesize)
```

Comment:
1. preceeding call won't finish until curl is over
2. if you want and need to check that original file and copied file are the same, please use md5sum or crc32 instead.
3. I'm not sure I understand what is  above function for since you run it only once

Reply:
Hey @izapolsk I had originally written this piece of code for one of my tests. I wanted to make sure the whole file was downloaded, so I wrote the function, I wasn't aware of md5sum or crc32 checks. I'll see if I can implement them.

Reply:
honestly, I doubt we need such check at all.

Reply:
Discussed with @valaparthvi and as per your suggestion. I have removed the check.
Thank you :)